x/pool-incentives/keeper/grpc_query.go

```diff
@@ -88,3 +90,27 @@ func (k Keeper) IncentivizedPools(ctx context.Context, _ *types.QueryIncentivize
 		IncentivizedPools: incentivizedPools,
 	}, nil
 }
+
+func (k Keeper) ExternalIncentiveGauges(ctx context.Context, req *types.QueryExternalIncentiveGaugesRequest) (*types.QueryExternalIncentiveGaugesResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+	store := sdkCtx.KVStore(k.storeKey)
+	prefixStore := prefix.NewStore(store, []byte("pool-incentives"))
+
+	iterator := prefixStore.Iterator(nil, nil)
+	defer iterator.Close()
+
+	poolGaugeIds := make(map[uint64]bool)
+	for ; iterator.Valid(); iterator.Next() {
+		poolGaugeIds[sdk.BigEndianToUint64(iterator.Value())] = true
+	}
+
+	allGauges := k.GetAllGauges(sdkCtx)
+	gauges := []incentivetypes.Gauge{}
+	for _, gauge := range allGauges {
+		if _, ok := poolGaugeIds[gauge.Id]; !ok {
+			gauges = append(gauges, gauge)
+		}
+	}
```

Comment:
This function as written will be iterating over all gauges, which is ~ 1500 state entries. I have a suspition that this may cause issues, but lets test.

Seems like the way to fix this would be in the future to index ActiveGauges separately, where Active was taken to me Active in the current sense, and having non-zero rewards. Is there a way to do this right now? I'm not sure. If not, we can merge the PR without this fix, and then flag this as something to fix in a separate issue.

Reply:
@ValarDragon There seems to be no other way now. I think we try it and fix it if there is a problem in future.